### PR TITLE
refactor: separation of debug-related functions from the main body

### PR DESCRIPTION
**Pull request by xygyo77**
Separation of debug-related functions from the main body
split some functions

### CODE REVIEW

File: src/mtt_checker/README.md [hunk 1]
```diff
@@ -34,24 +30,24 @@ options:
...
-                        periodic time default:100.0 (ms) use normal mode only
+                        periodic time default:100.0 (ms)
   -d time, --deadline time
                         deadline detect time default:200.0 (ms)
   -t name, --topic name
-                        topic name: Specify the target topic if there are
-                        multiple MTTs in the rosbag. default:
-                        for_tilde_interpolator_mtt
-
+                        topic name: Specify the target topic if there are multiple MTTs in the rosbag. default: for_tilde_interpolator_mtt
```
Comment (nabetetsu) [inline src/mtt_checker/README.md:37]:
```suggestion
                        topic name: Specify the target topic if there are multiple MTTs in the rosbag. Devault value is "for_tilde_interpolator_mtt".
```

File: src/mtt_checker/README.md [hunk 2]
```diff
@@ -34,24 +30,24 @@ options:
...
 ```
 
-出力結果(最終行辺り)  
+出力(結果は最終行辺り)  
 **シンプルモード**  
 MTT の stamp とその間隔のみで簡易的にデッドラインミスを検出する
 
-```bash
-ros2 run mtt_checker mtt_checker tp-ekf-pose-PREV-AWF.yaml -d 200 -m simple
+```
```
Comment (nabetetsu) [inline src/mtt_checker/README.md:44]:
```suggestion
```bash
```

File: src/tilde_timing_monitor/README.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ If you use the TILDE message tracking tag, the response time will be the time un
   - repo: <https://github.com/xygyo77/tilde-autoware.git>
   - branch: humble
 
-  ```bash
+  ```
```
Comment (nabetetsu) [inline src/tilde_timing_monitor/README.md:22]:
```suggestion
  ```bash
```

File: src/tilde_timing_monitor/README.md [hunk 4]
```diff
@@ -29,16 +29,39 @@ note: ndt_scan_matcher contains a unique change that outputs the latest EKF pose
 
 - repo: <https://github.com/tier4/tilde_lite.git>
 
-```bash
+```
```
Comment (nabetetsu) [inline src/tilde_timing_monitor/README.md:32]:
```suggestion
```bash
```

File: src/tilde_timing_monitor/src/tilde_timing_monitor_debug.cpp [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+
+#include "tilde_timing_monitor/tilde_timing_monitor_core.hpp"
+
+#define FMT_HEADER_ONLY
+#include <fmt/format.h>
+
+namespace tilde_timing_monitor
+{
+// const
+static const char * tm_command_topic = "tilde_timing_monitor_command";
```
Comment (nabetetsu) [inline src/tilde_timing_monitor/src/tilde_timing_monitor_debug.cpp:25]:
```suggestion
static const char * const tm_command_topic = "tilde_timing_monitor_command";
```

あるいは constexpr std::string_view など。

**Review by nabetetsu — COMMENTED** [review]
It may be better to remove MTT Checker from the PR to focus the refactoring in that PR.

File: src/tilde_timing_monitor/src/tilde_timing_monitor_debug.cpp (hunk 5 above)
Comment (xygyo77) [reply src/tilde_timing_monitor/src/tilde_timing_monitor_debug.cpp:25]: Is it ok to do "static constexpr char xx[] = "yy";" ?

File: src/tilde_timing_monitor/src/tilde_timing_monitor_core.cpp [hunk 6]
```diff
@@ -26,22 +28,13 @@
 namespace tilde_timing_monitor
 {
 // const
-static const char * version = "v0.02";
+static const char * version = "v0.12";
```
[the remark this answers is not in the document]
Comment (xygyo77) [reply src/tilde_timing_monitor/src/tilde_timing_monitor_core.cpp:31]: How about "static constexpr char version[] = "v0.12";" ?

**Comment by xygyo77** [comment]
> It may be better to remove MTT Checker from the PR to focus the refactoring in that PR.

surely

File: src/tilde_timing_monitor/README.md [hunk 7]
```diff
@@ -101,10 +124,10 @@ cp ~/colcon_ws/tilde_lite/src/tilde_timing_monitor/config/tilde_path_info.yaml .
...
 ```
 
-| param      | value          | content                       | default |
-| ---------- | -------------- | ----------------------------- | ------- |
-| mode       | by config file | Measurement path type         | test    |
-| statistics | bool           | statistics collection control | true    |
+| param      | value          | content               | default |
+| ---------- | -------------- | --------------------- | ------- |
+| mode       | by config file | Measurement path type | test    |
+| debug_ctrl | bool           | debug control         | true    |
```
Comment (nabetetsu) [inline src/tilde_timing_monitor/README.md:130]:
```suggestion
| debug | bool           | debug control         | true    |
```

Comment (nabetetsu) [reply src/tilde_timing_monitor/README.md:130]: I'm not sure but it looks `debug_ctrl` is used in the code for debug purpose but it is named `debug` as for external parameter.

Comment (nabetetsu) [reply src/tilde_timing_monitor/README.md:130]: I'm not sure but it looks `debug_ctrl` is used in the code for debug purpose but it is named `debug` as for external parameter.

Comment (xygyo77) [reply src/tilde_timing_monitor/README.md:130]: As you said, "debug" is correct.

**Review by nabetetsu — APPROVED** [review]

**Pull request merged**